### PR TITLE
Split virtual device and basic sharding tests

### PR DESCRIPTION
**Pull request by steventk-g**
When we run virtual device tests with basic sharding tests in the same process, we get crashes. When we run these tests in their own processes, they all pass. 

Since we currently only support configuring virtual device at the host process level (i.e. as an environment variable), we should have no loss of real world coverage separating these tests.

### CODE REVIEW

File: test/test_xla_virtual_device.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import os
```
Comment (JackCaoG) [inline test/test_xla_virtual_device.py:1]: I think we need to update run_test.sh. Can we also move all SPMD test into a separate dir like `test/spmd/`

File: torch_xla/test/test_xla_sharding_base.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from torch import nn
+import torch_xla
+import torch_xla.core.xla_model as xm
+import torch_xla.experimental.xla_sharding as xs
+from torch_xla.experimental.pjrt import using_pjrt
+
+
+@unittest.skipIf(not using_pjrt() or xm.get_xla_supported_devices("GPU"),
+                 f"Requires PJRT_DEVICE set to `TPU` or `CPU`.")
+class XlaShardingTest(unittest.TestCase):
```
Comment (JackCaoG) [inline torch_xla/test/test_xla_sharding_base.py:14]: I think it is better to put this file under `/test/spmd` instead of `torch_xla/test`

Comment (steventk-g) [reply torch_xla/test/test_xla_sharding_base.py:14]: As far as I can tell, we need to put it under `torch_xla` so that it can be imported by tests (like `test_util.py` in `torch_xla/test`. I could create `torch_xla/test/spmd` to make it more explicit.

Comment (JackCaoG) [reply torch_xla/test/test_xla_sharding_base.py:14]: I can you put import from python file at the same path, check https://github.com/pytorch/xla/blob/master/test/args_parse.py

Comment (steventk-g) [reply torch_xla/test/test_xla_sharding_base.py]: Done

**Review by JackCaoG — APPROVED** [review]

**Review by yeounoh — COMMENTED** [review]
Post-approval. LGTM.
